meson_cpu/x86/meson.build

```diff
@@ -1,228 +1,102 @@
-  group: ['AVX512ER', 'AVX512PF'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_knl.c')[0]
-)
-AVX512_KNM = mod_features.new(
-  'AVX512_KNM', 43, implies: AVX512_KNL,
-  args: ['-mavx5124fmaps', '-mavx5124vnniw', '-mavx512vpopcntdq'],
-  group: ['AVX5124FMAPS', 'AVX5124VNNIW', 'AVX512VPOPCNTDQ'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_knm.c')[0]
-)
-AVX512_SKX = mod_features.new(
-  'AVX512_SKX', 50, implies: AVX512CD,
-  args: ['-mavx512vl', '-mavx512bw', '-mavx512dq'],
-  group: ['AVX512VL', 'AVX512BW', 'AVX512DQ'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_skx.c')[0],
-  extra_tests: {
-    'AVX512BW_MASK': files(source_root + '/numpy/distutils/checks/extra_avx512bw_mask.c')[0],
-    'AVX512DQ_MASK': files(source_root + '/numpy/distutils/checks/extra_avx512dq_mask.c')[0]
-  }
-)
-AVX512_CLX = mod_features.new(
-  'AVX512_CLX', 51, implies: AVX512_SKX, args: '-mavx512vnni',
-  group: ['AVX512VNNI'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_clx.c')[0]
-)
-AVX512_CNL = mod_features.new(
-  'AVX512_CNL', 52, implies: AVX512_SKX,
-  args: ['-mavx512ifma', '-mavx512vbmi'],
-  group: ['AVX512IFMA', 'AVX512VBMI'],
-  test_code: files(source_root + '/numpy/distutils/checks/cpu_avx512_cnl.c')[0]
-)
+HWY_SSE4_FLAGS = ['-DHWY_WANT_SSE4', '-DHWY_DISABLE_PCLMUL_AES']
+# Use SSE for floating-point on x86-32 to ensure numeric consistency.
+# The x87 FPU's 80-bit internal precision causes unpredictable rounding
+# and overflow behavior when converting to smaller types. SSE maintains
+# strict 32/64-bit precision throughout all calculations.
+X86_64_V2_FLAGS = cpu_family == 'x86'? ['-mfpmath=sse'] : ['-mcx16']
+X86_64_V2_NAMES = cpu_family == 'x86'? [] : ['CX16']
+X86_V2 = mod_features.new(
+  'X86_V2', 1, args: ['-msse', '-msse2', '-msse3', '-mssse3', '-msse4.1', '-msse4.2',
+                      '-mpopcnt', '-msahf'] + X86_64_V2_FLAGS + HWY_SSE4_FLAGS,
+  # Adds compiler definitions `NPY_HAVE_SSE*` 
+  group: ['SSE', 'SSE2', 'SSE3', 'SSSE3', 'SSE41', 'SSE42', 'POPCNT', 'LAHF'] + X86_64_V2_NAMES,
+  detect: 'X86_V2',
+  test_code: files(current_dir + '/test_x86_v2.c')[0],
+)
+X86_V3 = mod_features.new(
+  'X86_V3', 10, implies: X86_V2, 
+  args: ['-mavx', '-mavx2', '-mfma', '-mbmi', '-mbmi2', '-mlzcnt', '-mf16c', '-mmovbe'],
+  group: ['AVX', 'AVX2', 'FMA3', 'BMI', 'BMI2', 'LZCNT', 'F16C', 'MOVBE'],
+  detect: 'X86_V3',
+  test_code: files(current_dir + '/test_x86_v3.c')[0],
+)
+X86_V4 = mod_features.new(
+  'X86_V4', 20, implies: X86_V3,
+  args: ['-mavx512f', '-mavx512cd', '-mavx512vl', '-mavx512bw', '-mavx512dq'],
+  group: ['AVX512F', 'AVX512CD', 'AVX512VL', 'AVX512BW', 'AVX512DQ', 'AVX512_SKX',
+          'AVX512F_REDUCE', 'AVX512BW_MASK', 'AVX512DQ_MASK'],
+  detect: 'X86_V4',
+  test_code: files(current_dir + '/test_x86_v4.c')[0],
+)
+if cpu_family == 'x86'
+  X86_V4.update(disable: 'not supported on x86-32')
```

Comment:
This disable needs to be backported too. We should not generate `AVX512` kernels on 32-bit systems.